Tella/Scenes/Settings/Views/Servers/Uwazi/Step2/UwaziServerAccessSelectionView.swift

```diff
@@ -14,80 +14,83 @@ struct UwaziServerAccessSelectionView: View {
-                                             isValid: .constant(true),action: {
-                        self.isLoginSelected = true
-                        self.isPublicInstance = false
-                    })
-                    .overlay( self.isLoginSelected ?
-                              RoundedRectangle(cornerRadius: 20)
-                        .stroke(.white, lineWidth: 4) : nil
-                    )
-                    TellaButtonView<AnyView>(title: LocalizableSettings.UwaziPublicInstance.localized,
-                                             nextButtonAction: .action,
-                                             isValid: .constant(true),action: {
-                        self.isLoginSelected = false
-                        self.isPublicInstance = true
-                    })
-                    .overlay( self.isPublicInstance ?
-                              RoundedRectangle(cornerRadius: 20)
-                        .stroke(.white, lineWidth: 4) : nil
-                    )
-                }.padding(EdgeInsets(top: 0, leading: 24, bottom: 0, trailing: 24))
+                    buttonView()
+                }
                 Spacer()
                 BottomLockView<AnyView>(isValid: $isButtonValid,
                                         nextButtonAction: .action,
                                         nextAction: {
-                    if isLoginSelected {
-                        let loginView = UwaziLoginView()
-                            .environmentObject(serversViewModel)
-                            .environmentObject(serverViewModel)
-                        navigateTo(destination: loginView)
-                    } else if isPublicInstance {
-                        let languageSelection = UwaziLanguageSelectionView(isPresented: .constant(true))
-                            //.environmentObject(SettingsViewModel(appModel: mainAppModel))
-                            .environmentObject(serversViewModel)
-                            .environmentObject(serverViewModel)
-                        navigateTo(destination: languageSelection)
-                    } else {
-
-                    }
+                    handleNavigation()
                 },
                 backAction: {
                     self.presentationMode.wrappedValue.dismiss()
                 })
-            }
+            }.padding(EdgeInsets(top: 0, leading: 16, bottom: 0, trailing: 16))
         }
-        // TODO: If there previous selection needed to be retained
-//        .onAppear {
-//            if let server = serverViewModel.currentServer {
-//                if server.username == "" {
-//                    self.isPublicInstance = true
-//                } else {
-//                    self.isLoginSelected = true
-//                }
-//            }
-//
-//        }
         .navigationBarBackButtonHidden(true)
 
     }
+    fileprivate func buttonView() -> VStack<TupleView<(TellaButtonView<AnyView>, TellaButtonView<AnyView>)>> {
```

Comment:
I think it's unuseful to to use  `TupleView` 
you can write just  `some View`

    fileprivate func buttonView() -> some View {
        VStack(spacing: 12) {
            TellaButtonView<AnyView>(title: LocalizableSettings.UwaziLogin.localized,
                                     nextButtonAction: .action,
                                     isOverlay: self.isLoginSelected,
                                     isValid: .constant(true),action: {
                self.isLoginSelected = true
                self.isPublicInstance = false
            })
            TellaButtonView<AnyView>(title: LocalizableSettings.UwaziPublicInstance.localized,
                                     nextButtonAction: .action,
                                     isOverlay: self.isPublicInstance,
                                     isValid: .constant(true),action: {
                self.isLoginSelected = false
                self.isPublicInstance = true
            })
        }

Reply:
Done